src/main/java/com/mycompany/myapp/service/tool/ToolService.java

```diff
@@ -0,0 +1,38 @@
+package com.mycompany.myapp.service.tool;
+
+import com.mycompany.myapp.domain.Pool;
+import com.mycompany.myapp.domain.Tool;
+import com.mycompany.myapp.repository.ToolRepository;
+import com.mycompany.myapp.service.pool.PoolService;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class ToolService {
+
+    private final ToolRepository toolRepository;
+
+    private final PoolService poolService;
+
+    public ToolService(ToolRepository toolRepository, PoolService poolService){
+        this.toolRepository = toolRepository;
+        this.poolService = poolService;
+    }
+
+    public List<Tool> getAdditionalTools(String poolId){
```

Comment:
I don't agree with the way that you are doing it. You are performing at least 3 requests to DB per call and one of it is `findAll` which basically takes all memory. I believe it can be done on DB itself either via view or function.